crates/bevy_asset/src/lib.rs

```diff
@@ -1986,4 +1986,95 @@ mod tests {
 
         app.world_mut().run_schedule(Update);
     }
+
+    #[test]
+    // CAUTION: Due to https://github.com/bevyengine/bevy/issues/18267, the
+    // test is expected to fail. The `#[should_panic] should be removed when the
+    // issue is fixed.
+    #[should_panic(expected = "assertion `left != right` failed")]
```

Comment:
I feel like marking this test as ignored is semantically clearer, and going to be easier to search for and fix later.